In [77]:
import pandas as pd
import numpy as np
import matplotlib as plt

In [78]:
def formata_valor(valor):
  return f"{valor:,.0f}".replace(',', '.')

In [79]:
path_vendas = "../data/Vendas.xlsx"
path_vendas_dez = "../data/Vendas_Dez.xlsx"
path_gerentes = "../data/Gerentes.xlsx"
df_vendas = pd.read_excel(path_vendas)
df_vendas_dez = pd.read_excel(path_vendas_dez)
df_gerentes = pd.read_excel(path_gerentes)

In [80]:
display(df_vendas.head(3))
display("Linhas / colunas", df_vendas.shape)

Código Venda       Data             ID Loja         Produto  Quantidade  \
0             1 2019-01-01  Iguatemi Esplanada  Sapato Estampa           1   
1             1 2019-01-01  Iguatemi Esplanada        Camiseta           2   
2             1 2019-01-01  Iguatemi Esplanada   Sapato Xadrez           1   

   Valor Unitário  Valor Final  
0             358          358  
1             180          360  
2             368          368

'Linhas / colunas'

(93910, 7)

In [81]:
print('Dados Estatísticos sobre valores')
display(df_vendas[['Valor Unitário', 'Valor Final']].describe().round(2))

Dados Estatísticos sobre valores


Valor Unitário  Valor Final
count        93910.00     93910.00
mean           191.73       414.86
std            145.22       434.85
min             30.00        30.00
25%            100.00       156.00
50%            155.00       274.00
75%            248.00       524.00
max            750.00      3750.00

In [82]:
display(df_vendas['ID Loja'].nunique())
display(df_vendas['Produto'].nunique())

25

120

As vendas contam com 25 lojas e 120 produtos diferentes em venda

In [83]:
total_vendas_por_lojas = df_vendas.groupby('ID Loja')['Valor Final'].sum().apply(formata_valor)
print('TOTAL VENDIDO POR UNIDADE')
display(total_vendas_por_lojas)

TOTAL VENDIDO POR UNIDADE


ID Loja
Bourbon Shopping SP                 1.600.260
Center Shopping Uberlândia          1.543.961
Iguatemi Campinas                   1.612.460
Iguatemi Esplanada                  1.596.339
Norte Shopping                      1.610.477
Novo Shopping Ribeirão Preto        1.578.522
Palladium Shopping Curitiba         1.600.499
Parque Dom Pedro Shopping           1.509.575
Passei das Águas Shopping           1.536.925
Ribeirão Shopping                   1.594.186
Rio Mar Recife                      1.582.897
Rio Mar Shopping Fortaleza          1.570.540
Salvador Shopping                   1.517.583
Shopping Barra                      1.549.955
Shopping Center Interlagos          1.536.665
Shopping Center Leste Aricanduva    1.596.714
Shopping Eldorado                   1.523.772
Shopping Ibirapuera                 1.527.153
Shopping Iguatemi Fortaleza         1.550.642
Shopping Midway Mall                1.485.172
Shopping Morumbi                    1.468.508
Shopping Recife           

In [84]:
total_produtos_vendidos = df_vendas.groupby('Produto')['Quantidade'].sum().apply(formata_valor)
print("TOTAL DE PRODUTOS VENDIDOS")
display(total_produtos_vendidos)

print("VALOR POR PRODUTO")
total_vendas_por_produto = df_vendas.groupby('Produto')['Valor Final'].sum().apply(formata_valor)
display(total_vendas_por_produto)

TOTAL DE PRODUTOS VENDIDOS


Produto
Bermuda             1.692
Bermuda Estampa     1.616
Bermuda Linho       1.893
Bermuda Liso        1.637
Bermuda Listrado    1.675
                    ...  
Tênis Estampa       1.663
Tênis Linho         1.713
Tênis Liso          1.656
Tênis Listrado      1.737
Tênis Xadrez        1.661
Name: Quantidade, Length: 120, dtype: object

VALOR POR PRODUTO


Produto
Bermuda             253.800
Bermuda Estampa     273.104
Bermuda Linho       369.135
Bermuda Liso        257.009
Bermuda Listrado    273.025
                     ...   
Tênis Estampa       425.728
Tênis Linho         503.622
Tênis Liso          440.496
Tênis Listrado      455.094
Tênis Xadrez        430.199
Name: Valor Final, Length: 120, dtype: object